include/deal.II/fe/mapping_fe_field.h

```diff
@@ -653,6 +656,19 @@ class MappingFEField : public Mapping<dim, spacedim>
   friend class MappingFEField;
 };
 
+/**
+ * @copydoc MappingFEField<dim, spacedim, VectorType>
+ *
+ * @deprecated Use MappingFEField<dim, spacedim, VectorType> instead.
+ */
+template <int dim, int spacedim, typename VectorType>
+class DEAL_II_DEPRECATED
+  MappingFEField<dim, spacedim, VectorType, DoFHandler<dim, spacedim>>
```

Comment:
I only added a explicit specialization for the basic `DoFHandler` as this is the only one that we explicitly instantiate.

Reply:
Hmmm... I would rather keep this fully templated on `DoFHandlerType` (and also specify for the other specialization that the last template argument is `void`).

Reply:
I like your suggestions and provided a patch 4bb1b81. Is this what you had in mind?

Reply:
With the fully templated `DoFHandlerType`, `doxygen` keeps warning about `Detected potential recursive class relation between class MappingFEField and base class MappingFEField< dim, spacedim, VectorType >!` at the definition of the deprecated class which causes the `indent` worker to fail.

Do you have an idea to circumvent this? Maybe place `#ifndef DOXYGEN ... #endif` around the class definition? I'll give it a try...

Reply:
That took some trial and error. We need a fifth template argument to have an explicit specialization for the fully templated `DoFHandlerType`. This way we can deprecate that particular specialization and leave the other non-deprecated. And `doxygen` also seems to be happy this time.

I did not specify the fifth template argument anywhere but in the base class declaration.